openmc/deplete/pool.py

```diff
@@ -72,45 +93,62 @@ def deplete(func, chain, x, rates, dt, matrix_func=None, transfer_rates=None,
     if transfer_rates is not None:
         # Calculate transfer rate terms as diagonal matrices
         transfers = map(chain.form_rr_term, repeat(transfer_rates),
-                        transfer_rates.burnable_mats)
+                        transfer_rates.local_mats)
         # Subtract transfer rate terms from Bateman matrices
         matrices = [matrix - transfer for (matrix, transfer) in zip(matrices,
                                                                     transfers)]
 
         if len(transfer_rates.index_transfer) > 0:
-            # Calculate transfer rate terms as diagonal matrices
-            transfer_pair = {
-                mat_pair: chain.form_rr_term(transfer_rates, mat_pair)
-                for mat_pair in transfer_rates.index_transfer
-            }
-
-            # Combine all matrices together in a single matrix of matrices
-            # to be solved in one go
-            n_rows = n_cols = len(transfer_rates.burnable_mats)
-            rows = []
-            for row in range(n_rows):
-                cols = []
-                for col in range(n_cols):
-                    mat_pair = (transfer_rates.burnable_mats[row],
-                                transfer_rates.burnable_mats[col])
-                    if row == col:
-                        # Fill the diagonals with the Bateman matrices
-                        cols.append(matrices[row])
-                    elif mat_pair in transfer_rates.index_transfer:
-                        # Fill the off-diagonals with the transfer pair matrices
-                        cols.append(transfer_pair[mat_pair])
-                    else:
-                        cols.append(None)
-
-                rows.append(cols)
-            matrix = bmat(rows)
-
-            # Concatenate vectors of nuclides in one
-            x_multi = np.concatenate([xx for xx in x])
-            x_result = func(matrix, x_multi, dt)
-
-            # Split back the nuclide vector result into the original form
-            x_result = np.split(x_result, np.cumsum([len(i) for i in x])[:-1])
+            # Gather all on comm.rank 0
+            matrices = comm.gather(matrices, root=0)
+            x = comm.gather(x, root=0)
```

Comment:
Default root is 0, so the second argument can be omitted
```suggestion
            matrices = comm.gather(matrices)
            x = comm.gather(x)
```